crdb_internal,pcr: add debugging vtables, builtin

dt opened the pull request.

dt committed:
pcr: pass consuming proc ID to subscribe

Release note: none.
Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/120734)
<!-- Reviewable:end -->

dt committed:
crdb_internal: add pcr producer debug table

Release note: none.
Epic: none.

dt committed:
streamproducer: collect debug info

Release note: none.
Epic: none.

dt committed:
sql: only test CTAS on 'supported' vtables

Release note: none.
Epic: none.

dt committed:
streamproducer: move lic checks into manager methods

Otherwise we cannot make a manager without a lic, but that might be annoying if we want non-lic
methods on a manager.

Release note: none.
Epic: none.

msbutler reviewed (approved):
LGTM!

dt commented:
TFTR!

bors r+

dt commented:
bors r+
